game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/retreat/DefensiveSubsRetreat.java

```diff
@@ -0,0 +1,100 @@
+package games.strategy.triplea.delegate.battle.steps.retreat;
+
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.SUBS_SUBMERGE;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.SUBS_WITHDRAW;
+import static games.strategy.triplea.delegate.battle.steps.BattleStep.Order.SUB_DEFENSIVE_RETREAT_AFTER_BATTLE;
+import static games.strategy.triplea.delegate.battle.steps.BattleStep.Order.SUB_DEFENSIVE_RETREAT_BEFORE_BATTLE;
+
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.MustFightBattle.RetreatType;
+import games.strategy.triplea.delegate.battle.steps.BattleStep;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import org.triplea.java.collections.CollectionUtils;
+
+@AllArgsConstructor
+public class DefensiveSubsRetreat implements BattleStep {
+
+  private static final long serialVersionUID = 1249467218938096244L;
+
+  protected final BattleState battleState;
+
+  protected final BattleActions battleActions;
+
+  @Override
+  public List<String> getNames() {
+    if (!isEvaderPresent() || !isRetreatPossible()) {
+      return List.of();
+    }
+
+    if (getOrder() == SUB_DEFENSIVE_RETREAT_BEFORE_BATTLE && isDestroyerPresent()) {
+      // only check for destroyers if subs can retreat before battle
+      // because the destroyer could be killed during the battle which would
+      // allow the sub to withdraw at the end of the battle
+      return List.of();
+    }
+
+    if (Properties.getSubmersibleSubs(battleState.getGameData())) {
+      return List.of(battleState.getDefender().getName() + SUBS_SUBMERGE);
```

Comment:
nit, any thoughts to use a ternary here to isolate the different parts of the 'if/else' block as much as possible?

EG:
```
String stepName =  battleState.getDefender().getName() + (Properties.getSubmersibleSubs(battleState.getGameData() ? SUBS_SUBMERGE : SUBS_WITHDRAW);
return List.of(stepName);
```